api.planx.uk/types.ts

```diff
@@ -61,53 +61,50 @@ export interface LowCalSession {
   };
 }
 
+interface MinimumNotifyPersonalisation {
+  emailReplyToId: string;
+  expiryDate?: string;
+  helpEmail: string;
+  helpOpeningHours: string;
+  helpPhone: string;
+  serviceLink?: string;
+  serviceName?: string;
+  sessionId?: string;
+}
+
+interface SaveAndReturnNotifyPersonalisation extends MinimumNotifyPersonalisation {
+  address?: any;
+  projectType?: string;
+  resumeLink?: string;
+  teamName: string;
+}
+
 export interface SaveAndReturnNotifyConfig {
-  personalisation: {
-    address?: any;
-    emailReplyToId: string;
-    expiryDate?: string;
-    helpEmail: string;
-    helpOpeningHours: string;
-    helpPhone: string;
-    id?: string;
-    projectType?: string;
-    resumeLink?: string;
-    serviceLink?: string;
-    serviceName?: string;
-    teamName: string;
-  };
+  personalisation: SaveAndReturnNotifyPersonalisation;
+}
+
+interface EmailSubmissionNotifyPersonalisation extends MinimumNotifyPersonalisation {
+  applicantEmail: string;
+  downloadLink: string;
 }
 
 export interface EmailSubmissionNotifyConfig {
-  personalisation: {
-    applicantEmail: string;
-    downloadLink: string;
-    emailReplyToId: string;
-    expiryDate?: string;
-    id?: string;
-    serviceName: string;
-    sessionId: string;
-  };
+  personalisation: EmailSubmissionNotifyPersonalisation;
+}
+
+interface InviteToPayNotifyPersonalisation extends MinimumNotifyPersonalisation {
```

Comment:
nit: Since this is an internal data structure (one we control), and since this is the exact set of data fields required for the template, would this be better modeled as a `type` (with fewer optional fields if possible). An `interface` is a bit looser than a `type` but I think we're aiming at quite strong typing for email templates.

Reply:
Thanks for highlighting this distinction - updated `MinimumNotifyPersonalisation` to be a `type`, the rest are still interfaces to make use of `extend` :+1: